master/buildbot/test/fake/reactor.py

```diff
@@ -79,7 +79,10 @@ def callInThreadWithCallback(self, onResult, func, *args, **kw):
         self.calls += 1
         try:
             result = func(*args, **kw)
-        except:
+        except:  # pylint: disable=bare-except
+            # We catch *everything* here, since normally this code would be
+            # running in a thread, where there is nothing that will catch
+            # error.
```

Comment:
Shouldn't we treat keyboard interrupt special case here, to be able to properly cancel a trial -u?

http://effbot.org/zone/stupid-exceptions-keyboardinterrupt.htm